#VRML_SIM R2022a utf8

PROTO MCUTimingGate [
  field SFVec3f translation 0.3 0 0
  field SFRotation rotation 0 0 1 0
  field SFString{"practice", "competition"} mode "practice"
  field SFInt32 laps 5
]
{
  Robot {
    translation IS translation
    rotation IS rotation
    children [
      DEF GATE Shape {
        appearance PBRAppearance {
          baseColor 0.9 0.9 0.9
          roughness 1
          metalness 0
          name "timing_gate"
        }
        geometry IndexedFaceSet {
          coord Coordinate {
            point [
              -0.1 0.4 -0.15, -0.1 0.40827 -0.384823, -0.1 0.369408 -0.393709, -0.1 0.691863 -0.0790496, -0.1 0.682029 -0.117683, -0.1 0.446055 -0.372113, -0.1 0.1 0.45, -0.1 0.1 0.346718, -0.1 0 0.45, -0.1 0.668398 -0.155145, -0.1 0.482387 -0.355704, -0.1 0.651106 -0.191065, -0.1 0.516905 -0.335761, -0.1 0.630324 -0.225084, -0.1 0 0.15, -0.1 0.549267 -0.312481, -0.1 0.606259 -0.256867, -0.1 0.57915 -0.286096, -0.1 0.135486 0.364884, -0.1 0.4 0.15, -0.1 0.172603 0.379427, -0.1 0.210984 0.390204, -0.1 0.250247 0.397106, -0.1 0.290002 0.400066, -0.1 0.329855 0.399055, -0.1 0.369408 0.394081, -0.1 0.40827 0.385195, -0.1 0.446055 0.372485, -0.1 0.482387 0.356077, -0.1 0.516905 0.336134, -0.1 0.549267 0.312854, -0.1 0.57915 0.286468, -0.1 0.606259 0.257239, -0.1 0.630324 0.225457, -0.1 0.1 -0.346346, -0.1 0 -0.45, -0.1 0 -0.15, -0.1 0.651106 0.191437, -0.1 0.668398 0.155517, -0.1 0.1 -0.45, -0.1 0.135486 -0.364511, -0.1 0.172603 -0.379055, -0.1 0.682029 0.118055, -0.1 0.210984 -0.389831, -0.1 0.691863 0.0794219, -0.1 0.250247 -0.396734, -0.1 0.697803 0.0400017, -0.1 0.290002 -0.399694, -0.1 0.699789 0.000186132, -0.1 0.329855 -0.398682, -0.1 0.697803 -0.0396295, 0.1 0.1 -0.45, 0.1 0 -0.45, -0.1 0 -0.45, -0.1 0.1 -0.45, 0.1 0 -0.45, 0.1 0 -0.15, -0.1 0 -0.15, -0.1 0 -0.45, 0.1 0 -0.15, 0.1 0.4 -0.15, -0.1 0.4 -0.15, -0.1 0 -0.15, 0.1 0.4 -0.15, 0.1 0.4 0.15, -0.1 0.4 0.15, -0.1 0.4 -0.15, 0.1 0.4 0.15, 0.1 0 0.15, -0.1 0 0.15, -0.1 0.4 0.15, 0.1 0 0.15, 0.1 0 0.45, -0.1 0 0.45, -0.1 0 0.15, 0.1 0 0.45, 0.1 0.1 0.45, -0.1 0.1 0.45, -0.1 0 0.45, 0.1 0.1 0.45, 0.1 0.1 0.346718, -0.1 0.1 0.346718, -0.1 0.1 0.45, 0.1 0.699789 0.000186132, -0.1 0.699789 0.000186132, -0.1 0.697803 0.0400017, 0.1 0.699789 0.000186132, 0.1 0.697803 -0.0396295, -0.1 0.699789 0.000186132, 0.1 0.697803 0.0400017, -0.1 0.697803 0.0400017, -0.1 0.691863 0.0794219, 0.1 0.697803 0.0400017, 0.1 0.691863 0.0794219, -0.1 0.691863 0.0794219, -0.1 0.682029 0.118055, 0.1 0.691863 0.0794219, 0.1 0.135486 -0.364511, 0.1 0.1 -0.346346, -0.1 0.1 -0.346346, 0.1 0.135486 -0.364511, -0.1 0.135486 -0.364511, -0.1 0.172603 -0.379055, -0.1 0.135486 -0.364511, 0.1 0.682029 0.118055, -0.1 0.682029 0.118055, -0.1 0.668398 0.155517, 0.1 0.682029 0.118055, 0.1 0.172603 -0.379055, -0.1 0.172603 -0.379055, -0.1 0.210984 -0.389831, 0.1 0.172603 -0.379055, 0.1 0.668398 0.155517, -0.1 0.668398 0.155517, -0.1 0.651106 0.191437, 0.1 0.668398 0.155517, 0.1 0.210984 -0.389831, 0.1 0.210984 -0.389831, -0.1 0.210984 -0.389831, -0.1 0.250247 -0.396734, 0.1 0.651106 0.191437, -0.1 0.651106 0.191437, -0.1 0.630324 0.225457, 0.1 0.651106 0.191437, 0.1 0.250247 -0.396734, 0.1 0.250247 -0.396734, -0.1 0.250247 -0.396734, -0.1 0.290002 -0.399694, 0.1 0.630324 0.225457, -0.1 0.630324 0.225457, -0.1 0.606259 0.257239, 0.1 0.630324 0.225457, 0.1 0.290002 -0.399694, 0.1 0.290002 -0.399694, -0.1 0.290002 -0.399694, -0.1 0.329855 -0.398682, 0.1 0.606259 0.257239, -0.1 0.606259 0.257239, -0.1 0.57915 0.286468, 0.1 0.606259 0.257239, 0.1 0.329855 -0.398682, 0.1 0.329855 -0.398682, -0.1 0.329855 -0.398682, -0.1 0.369408 -0.393709, 0.1 0.57915 0.286468, 0.1 0.57915 0.286468, -0.1 0.57915 0.286468, -0.1 0.549267 0.312854, 0.1 0.369408 -0.393709, 0.1 0.369408 -0.393709, -0.1 0.369408 -0.393709, -0.1 0.40827 -0.384823, 0.1 0.549267 0.312854, 0.1 0.549267 0.312854, -0.1 0.549267 0.312854, -0.1 0.516905 0.336134, 0.1 0.40827 -0.384823, 0.1 0.40827 -0.384823, -0.1 0.40827 -0.384823, -0.1 0.446055 -0.372113, 0.1 0.516905 0.336134, 0.1 0.516905 0.336134, -0.1 0.516905 0.336134, -0.1 0.482387 0.356077, 0.1 0.446055 -0.372113, 0.1 0.446055 -0.372113, -0.1 0.446055 -0.372113, -0.1 0.482387 -0.355704, 0.1 0.482387 0.356077, 0.1 0.482387 0.356077, -0.1 0.482387 0.356077, -0.1 0.446055 0.372485, 0.1 0.482387 -0.355704, 0.1 0.482387 -0.355704, -0.1 0.482387 -0.355704, -0.1 0.516905 -0.335761, 0.1 0.446055 0.372485, 0.1 0.446055 0.372485, -0.1 0.446055 0.372485, -0.1 0.40827 0.385195, 0.1 0.516905 -0.335761, 0.1 0.516905 -0.335761, -0.1 0.516905 -0.335761, -0.1 0.549267 -0.312481, 0.1 0.40827 0.385195, 0.1 0.40827 0.385195, -0.1 0.40827 0.385195, -0.1 0.369408 0.394081, 0.1 0.549267 -0.312481, 0.1 0.549267 -0.312481, -0.1 0.549267 -0.312481, -0.1 0.57915 -0.286096, 0.1 0.369408 0.394081, 0.1 0.369408 0.394081, -0.1 0.369408 0.394081, -0.1 0.329855 0.399055, 0.1 0.57915 -0.286096, 0.1 0.57915 -0.286096, -0.1 0.57915 -0.286096, -0.1 0.606259 -0.256867, 0.1 0.329855 0.399055, 0.1 0.329855 0.399055, -0.1 0.329855 0.399055, -0.1 0.290002 0.400066, 0.1 0.606259 -0.256867, 0.1 0.606259 -0.256867, -0.1 0.606259 -0.256867, -0.1 0.630324 -0.225084, 0.1 0.290002 0.400066, 0.1 0.290002 0.400066, -0.1 0.290002 0.400066, -0.1 0.250247 0.397106, 0.1 0.630324 -0.225084, 0.1 0.630324 -0.225084, -0.1 0.630324 -0.225084, -0.1 0.651106 -0.191065, 0.1 0.250247 0.397106, 0.1 0.250247 0.397106, -0.1 0.250247 0.397106, -0.1 0.210984 0.390204, 0.1 0.651106 -0.191065, 0.1 0.651106 -0.191065, -0.1 0.651106 -0.191065, -0.1 0.668398 -0.155145, 0.1 0.210984 0.390204, 0.1 0.210984 0.390204, -0.1 0.210984 0.390204, -0.1 0.172603 0.379427, 0.1 0.172603 0.379427, -0.1 0.172603 0.379427, -0.1 0.135486 0.364884, 0.1 0.668398 -0.155145, 0.1 0.668398 -0.155145, -0.1 0.668398 -0.155145, -0.1 0.682029 -0.117683, 0.1 0.172603 0.379427, 0.1 0.682029 -0.117683, 0.1 0.135486 0.364884, -0.1 0.135486 0.364884, -0.1 0.1 0.346718, 0.1 0.135486 0.364884, 0.1 0.682029 -0.117683, -0.1 0.682029 -0.117683, -0.1 0.691863 -0.0790496, 0.1 0.1 0.346718, 0.1 0.691863 -0.0790496, -0.1 0.691863 -0.0790496, -0.1 0.697803 -0.0396295, 0.1 0.691863 -0.0790496, -0.1 0.697803 -0.0396295, 0.1 0.697803 -0.0396295, 0.1 0.1 -0.45, -0.1 0.1 -0.45, -0.1 0.1 -0.346346, 0.1 0.1 -0.346346, 0.1 0.40827 -0.384823, 0.1 0.4 -0.15, 0.1 0.369408 -0.393709, 0.1 0.682029 -0.117683, 0.1 0.691863 -0.0790496, 0.1 0.446055 -0.372113, 0.1 0.1 0.346718, 0.1 0.1 0.45, 0.1 0 0.45, 0.1 0.668398 -0.155145, 0.1 0.482387 -0.355704, 0.1 0.651106 -0.191065, 0.1 0.516905 -0.335761, 0.1 0.630324 -0.225084, 0.1 0 0.15, 0.1 0.549267 -0.312481, 0.1 0.606259 -0.256867, 0.1 0.57915 -0.286096, 0.1 0.4 0.15, 0.1 0.549267 0.312854, 0.1 0.516905 0.336134, 0.1 0.482387 0.356077, 0.1 0.446055 0.372485, 0.1 0.40827 0.385195, 0.1 0.369408 0.394081, 0.1 0.329855 0.399055, 0.1 0.290002 0.400066, 0.1 0.250247 0.397106, 0.1 0.210984 0.390204, 0.1 0.172603 0.379427, 0.1 0.135486 0.364884, 0.1 0.57915 0.286468, 0.1 0.606259 0.257239, 0.1 0.630324 0.225457, 0.1 0 -0.45, 0.1 0.1 -0.346346, 0.1 0 -0.15, 0.1 0.651106 0.191437, 0.1 0.668398 0.155517, 0.1 0.135486 -0.364511, 0.1 0.1 -0.45, 0.1 0.172603 -0.379055, 0.1 0.682029 0.118055, 0.1 0.210984 -0.389831, 0.1 0.691863 0.0794219, 0.1 0.250247 -0.396734, 0.1 0.697803 0.0400017, 0.1 0.290002 -0.399694, 0.1 0.699789 0.000186132, 0.1 0.329855 -0.398682, 0.1 0.697803 -0.0396295
            ]
          }
          normal Normal {
            vector [
              -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, -1 0 0, 0 0 -1, 0 0 -1, 0 0 -1, 0 0 -1, 0 -1 0, 0 -1 0, 0 -1 0, 0 -1 0, 0 0 1, 0 0 1, 0 0 1, 0 0 1, 0 -1 0, 0 -1 0, 0 -1 0, 0 -1 0, 0 0 -1, 0 0 -1, 0 0 -1, 0 0 -1, 0 -1 0, 0 -1 0, 0 -1 0, 0 -1 0, 0 0 1, 0 0 1, 0 0 1, 0 0 1, 0 1 0, 0 1 0, 0 1 0, 0 1 0, 0 0.998758 0.0498323, 0 0.998758 0.0498323, 0 0.998758 0.0498323, 0 0.998758 -0.0498323, 0 0.998758 -0.0498323, 0 0.998758 -0.0498323, 0 0.988837 0.148999, 0 0.988837 0.148999, 0 0.988837 0.148999, 0 0.998758 0.0498323, 0 0.969095 0.246687, 0 0.969095 0.246687, 0 0.969095 0.246687, 0 0.988837 0.148999, 0 -0.455682 -0.890143, 0 -0.455682 -0.890143, 0 -0.455682 -0.890143, 0 -0.364815 -0.93108, 0 -0.364815 -0.93108, 0 -0.364815 -0.93108, 0 -0.455682 -0.890143, 0 0.939727 0.341924, 0 0.939727 0.341924, 0 0.939727 0.341924, 0 0.969095 0.246687, 0 -0.270325 -0.962769, 0 -0.270325 -0.962769, 0 -0.270325 -0.962769, 0 -0.364815 -0.93108, 0 0.901025 0.433767, 0 0.901025 0.433767, 0 0.901025 0.433767, 0 0.939727 0.341924, 0 -0.270325 -0.962769, 0 -0.173148 -0.984896, 0 -0.173148 -0.984896, 0 -0.173148 -0.984896, 0 0.853374 0.521299, 0 0.853374 0.521299, 0 0.853374 0.521299, 0 0.901025 0.433767, 0 -0.173148 -0.984896, 0 -0.074254 -0.997239, 0 -0.074254 -0.997239, 0 -0.074254 -0.997239, 0 0.797246 0.603654, 0 0.797246 0.603654, 0 0.797246 0.603654, 0 0.853374 0.521299, 0 -0.074254 -0.997239, 0 0.0253793 -0.999678, 0 0.0253793 -0.999678, 0 0.0253793 -0.999678, 0 0.7332 0.680013, 0 0.7332 0.680013, 0 0.7332 0.680013, 0 0.797246 0.603654, 0 0.0253793 -0.999678, 0 0.124761 -0.992187, 0 0.124761 -0.992187, 0 0.124761 -0.992187, 0 0.7332 0.680013, 0 0.66187 0.749618, 0 0.66187 0.749618, 0 0.66187 0.749618, 0 0.124761 -0.992187, 0 0.222902 -0.974841, 0 0.222902 -0.974841, 0 0.222902 -0.974841, 0 0.66187 0.749618, 0 0.583967 0.811777, 0 0.583967 0.811777, 0 0.583967 0.811777, 0 0.222902 -0.974841, 0 0.31883 -0.947812, 0 0.31883 -0.947812, 0 0.31883 -0.947812, 0 0.583967 0.811777, 0 0.500263 0.865874, 0 0.500263 0.865874, 0 0.500263 0.865874, 0 0.31883 -0.947812, 0 0.41159 -0.911369, 0 0.41159 -0.911369, 0 0.41159 -0.911369, 0 0.500263 0.865874, 0 0.411591 0.911369, 0 0.411591 0.911369, 0 0.411591 0.911369, 0 0.41159 -0.911369, 0 0.500263 -0.865874, 0 0.500263 -0.865874, 0 0.500263 -0.865874, 0 0.411591 0.911369, 0 0.31883 0.947812, 0 0.31883 0.947812, 0 0.31883 0.947812, 0 0.500263 -0.865874, 0 0.583967 -0.811777, 0 0.583967 -0.811777, 0 0.583967 -0.811777, 0 0.31883 0.947812, 0 0.222902 0.974841, 0 0.222902 0.974841, 0 0.222902 0.974841, 0 0.583967 -0.811777, 0 0.661871 -0.749618, 0 0.661871 -0.749618, 0 0.661871 -0.749618, 0 0.222902 0.974841, 0 0.124761 0.992187, 0 0.124761 0.992187, 0 0.124761 0.992187, 0 0.661871 -0.749618, 0 0.7332 -0.680013, 0 0.7332 -0.680013, 0 0.7332 -0.680013, 0 0.124761 0.992187, 0 0.0253793 0.999678, 0 0.0253793 0.999678, 0 0.0253793 0.999678, 0 0.7332 -0.680013, 0 0.797246 -0.603654, 0 0.797246 -0.603654, 0 0.797246 -0.603654, 0 0.0253793 0.999678, 0 -0.0742532 0.997239, 0 -0.0742532 0.997239, 0 -0.0742532 0.997239, 0 0.797246 -0.603654, 0 0.853374 -0.521299, 0 0.853374 -0.521299, 0 0.853374 -0.521299, 0 -0.0742532 0.997239, 0 -0.173149 0.984896, 0 -0.173149 0.984896, 0 -0.173149 0.984896, 0 0.853374 -0.521299, 0 0.901025 -0.433767, 0 0.901025 -0.433767, 0 0.901025 -0.433767, 0 -0.173149 0.984896, 0 -0.270324 0.962769, 0 -0.270324 0.962769, 0 -0.270324 0.962769, 0 -0.364816 0.93108, 0 -0.364816 0.93108, 0 -0.364816 0.93108, 0 0.901025 -0.433767, 0 0.939727 -0.341924, 0 0.939727 -0.341924, 0 0.939727 -0.341924, 0 -0.270324 0.962769, 0 0.939727 -0.341924, 0 -0.455682 0.890143, 0 -0.455682 0.890143, 0 -0.455682 0.890143, 0 -0.364816 0.93108, 0 0.969095 -0.246687, 0 0.969095 -0.246687, 0 0.969095 -0.246687, 0 -0.455682 0.890143, 0 0.988837 -0.148999, 0 0.988837 -0.148999, 0 0.988837 -0.148999, 0 0.969095 -0.246687, 0 0.998758 -0.0498323, 0 0.988837 -0.148999, 0 1 0, 0 1 0, 0 1 0, 0 1 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0, 1 0 0
            ]
          }
          coordIndex [
            0, 1, 2, -1, 3, 4, 0, -1, 0, 5, 1, -1, 6, 7, 8, -1, 4, 9, 0, -1, 0, 10, 5, -1, 9, 11, 0, -1, 0, 12, 10, -1, 11, 13, 0, -1, 7, 14, 8, -1, 0, 15, 12, -1, 13, 16, 0, -1, 0, 17, 15, -1, 16, 17, 0, -1, 18, 19, 7, -1, 20, 19, 18, -1, 21, 19, 20, -1, 22, 19, 21, -1, 23, 19, 22, -1, 24, 19, 23, -1, 25, 19, 24, -1, 26, 19, 25, -1, 27, 19, 26, -1, 28, 19, 27, -1, 29, 19, 28, -1, 30, 19, 29, -1, 7, 19, 14, -1, 30, 31, 19, -1, 31, 32, 19, -1, 32, 33, 19, -1, 34, 35, 36, -1, 33, 37, 19, -1, 37, 38, 19, -1, 34, 39, 35, -1, 34, 0, 40, -1, 36, 0, 34, -1, 0, 41, 40, -1, 38, 42, 19, -1, 0, 43, 41, -1, 42, 44, 19, -1, 0, 45, 43, -1, 44, 46, 19, -1, 0, 47, 45, -1, 19, 48, 0, -1, 46, 48, 19, -1, 0, 49, 47, -1, 48, 50, 0, -1, 0, 2, 49, -1, 50, 3, 0, -1, 51, 52, 53, -1, 51, 53, 54, -1, 55, 56, 57, -1, 55, 57, 58, -1, 59, 60, 61, -1, 59, 61, 62, -1, 63, 64, 65, -1, 63, 65, 66, -1, 67, 68, 69, -1, 67, 69, 70, -1, 71, 72, 73, -1, 71, 73, 74, -1, 75, 76, 77, -1, 75, 77, 78, -1, 79, 80, 81, -1, 79, 81, 82, -1, 83, 84, 85, -1, 86, 87, 88, -1, 89, 90, 91, -1, 92, 83, 85, -1, 93, 94, 95, -1, 96, 89, 91, -1, 97, 98, 99, -1, 100, 101, 102, -1, 97, 99, 103, -1, 104, 105, 106, -1, 107, 93, 95, -1, 108, 109, 110, -1, 111, 100, 102, -1, 112, 113, 114, -1, 115, 104, 106, -1, 116, 108, 110, -1, 117, 118, 119, -1, 120, 121, 122, -1, 123, 112, 114, -1, 124, 117, 119, -1, 125, 126, 127, -1, 128, 129, 130, -1, 131, 120, 122, -1, 132, 125, 127, -1, 133, 134, 135, -1, 136, 137, 138, -1, 139, 128, 130, -1, 140, 133, 135, -1, 141, 142, 143, -1, 144, 136, 138, -1, 145, 146, 147, -1, 148, 141, 143, -1, 149, 150, 151, -1, 152, 145, 147, -1, 153, 154, 155, -1, 156, 149, 151, -1, 157, 158, 159, -1, 160, 153, 155, -1, 161, 162, 163, -1, 164, 157, 159, -1, 165, 166, 167, -1, 168, 161, 163, -1, 169, 170, 171, -1, 172, 165, 167, -1, 173, 174, 175, -1, 176, 169, 171, -1, 177, 178, 179, -1, 180, 173, 175, -1, 181, 182, 183, -1, 184, 177, 179, -1, 185, 186, 187, -1, 188, 181, 183, -1, 189, 190, 191, -1, 192, 185, 187, -1, 193, 194, 195, -1, 196, 189, 191, -1, 197, 198, 199, -1, 200, 193, 195, -1, 201, 202, 203, -1, 204, 197, 199, -1, 205, 206, 207, -1, 208, 201, 203, -1, 209, 210, 211, -1, 212, 205, 207, -1, 213, 214, 215, -1, 216, 209, 211, -1, 217, 218, 219, -1, 220, 213, 215, -1, 221, 222, 223, -1, 224, 217, 219, -1, 225, 226, 227, -1, 228, 229, 230, -1, 231, 221, 223, -1, 232, 233, 234, -1, 235, 225, 227, -1, 236, 232, 234, -1, 237, 238, 239, -1, 240, 228, 230, -1, 241, 242, 243, -1, 244, 237, 239, -1, 245, 246, 247, -1, 248, 241, 243, -1, 87, 249, 88, -1, 250, 245, 247, -1, 251, 252, 253, -1, 254, 251, 253, -1, 255, 256, 257, -1, 258, 259, 256, -1, 260, 256, 255, -1, 261, 262, 263, -1, 264, 258, 256, -1, 265, 256, 260, -1, 266, 264, 256, -1, 267, 256, 265, -1, 268, 266, 256, -1, 269, 261, 263, -1, 270, 256, 267, -1, 271, 268, 256, -1, 272, 271, 256, -1, 272, 256, 270, -1, 273, 274, 275, -1, 273, 275, 276, -1, 273, 276, 277, -1, 273, 277, 278, -1, 273, 278, 279, -1, 273, 279, 280, -1, 273, 280, 281, -1, 273, 281, 282, -1, 273, 282, 283, -1, 273, 283, 284, -1, 273, 284, 285, -1, 273, 285, 261, -1, 273, 261, 269, -1, 286, 274, 273, -1, 287, 286, 273, -1, 288, 287, 273, -1, 289, 290, 291, -1, 292, 288, 273, -1, 293, 292, 273, -1, 256, 290, 294, -1, 256, 291, 290, -1, 295, 290, 289, -1, 296, 256, 294, -1, 297, 293, 273, -1, 298, 256, 296, -1, 299, 297, 273, -1, 300, 256, 298, -1, 301, 299, 273, -1, 302, 256, 300, -1, 303, 273, 256, -1, 303, 301, 273, -1, 304, 256, 302, -1, 305, 303, 256, -1, 257, 256, 304, -1, 259, 305, 256, -1
          ]
        }
      }
      Display {
        translation 0.1 0.57 0
        rotation 0 -1 0 1.5708
        children [
          Shape {
            appearance PBRAppearance {
              baseColorMap ImageTexture {
              }
              roughness 1
              metalness 0
            }
            geometry IndexedFaceSet {
              coord Coordinate {
                point [
                  0.12 0.08 -0.01
                  0.12 -0.08 -0.01
                  -0.12 -0.08 -0.01
                  -0.12 0.08 -0.01
                ]
              }
              texCoord TextureCoordinate {
                point [
                  0 1
                  0 0
                  1 0
                  1 1
                ]
              }
              coordIndex [
                0, 1, 2, 3
              ]
              texCoordIndex [
                0, 1, 2, 3
              ]
            }
          }
          Shape {
            appearance PBRAppearance {
              baseColor 0.3 0.3 0.3
            }
            geometry Box {
              size 0.26, 0.18, 0.005
            }
          }
        ]
        name "display_front"
        width 256
        height 128
      }
      Display {
        translation -0.1 0.57 0
        rotation 0 1 0 1.5708
        children [
          Shape {
            appearance PBRAppearance {
              baseColorMap ImageTexture {
              }
              roughness 1
              metalness 0
            }
            geometry IndexedFaceSet {
              coord Coordinate {
                point [
                  0.12 0.08 -0.01
                  0.12 -0.08 -0.01
                  -0.12 -0.08 -0.01
                  -0.12 0.08 -0.01
                ]
              }
              texCoord TextureCoordinate {
                point [
                  0 1
                  0 0
                  1 0
                  1 1
                ]
              }
              coordIndex [
                0, 1, 2, 3
              ]
              texCoordIndex [
                0, 1, 2, 3
              ]
            }
          }
          Shape {
            appearance PBRAppearance {
              baseColor 0.3 0.3 0.3
            }
            geometry Box {
              size 0.26, 0.18, 0.005
            }
          }
        ]
        name "display_back"
        width 256
        height 128
      }
      %{
        for i = 0, 4 do
          local name = 'led_front_' .. i
      }%
      LED {
        translation 0.1 0.43 %{=0.1 - i * 0.05}%
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 1 1 1
              emissiveIntensity 50
            }
            geometry Sphere {
              radius 0.005
            }
          }
        ]
        name "%{=name}%"
      }
      %{ end }%
      %{
        for i = 0, 4 do
          name = 'led_back_' .. i
      }%
      LED {
        translation -0.1 0.43 %{=-0.1 + i * 0.05}%
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 1 1 1
              emissiveIntensity 50
            }
            geometry Sphere {
              radius 0.005
            }
          }
        ]
        name "%{=name}%"
      }
      %{ end }%
      DistanceSensor {
        translation 0 0.035 -0.14
        rotation 0 1 0 -1.5708
        name "distance_sensor"
      }
    ]
    name "timing_gate"
    boundingObject USE GATE
    supervisor TRUE
    controller "race"
  }

}
